pkg/dependency/parser/golang/binary/parse.go

```diff
@@ -154,16 +163,46 @@ func (p *Parser) ParseLDFlags(name string, flags []string) string {
 		// -X "main.version=1.0.0"
 		key = strings.TrimLeft(key, `'`)
 		val = strings.TrimRight(val, `'`)
-		if isValidXKey(key) && isValidSemVer(val) {
-			return val
+		if isVersionXKey(key) && isValidSemVer(val) {
```

Comment:
As I commented here, I think all keys should start with the module name. Can we miss any versions with this logic?

```suggestion
		if !strings.HasPrefix(key, name) {
			continue
		}
		if isVersionXKey(key) && isValidSemVer(val) {
```

Reply:
In this case we will miss `main.Version=v1.0.0` (e.g. https://www.digitalocean.com/community/tutorials/using-ldflags-to-set-version-information-for-go-applications#using-ldflags-with-go-build)

We also used this ldflag in Trivy - https://github.com/aquasecurity/trivy/blob/f1052799894cc8a8480ff28e1c717a9d811876a2/goreleaser.yml#L9

Reply:
I didn't remember the variable can be passed without the module name. Thanks!